src/plugins/part.c

```diff
@@ -1726,6 +1726,78 @@ gboolean bd_part_delete_part (const gchar *disk, const gchar *part, GError **err
     return TRUE;
 }
 
+/* get maximal size for partition when resizing
+ * this is a simplified copy of 'resize_get_last_possible' function from
+ * libfdisk which is unfortunately not public
+ */
+static gboolean get_max_part_size (struct fdisk_table *tb, guint partno, guint64 *max_size, GError **error) {
+    struct fdisk_partition *pa = NULL;
+    struct fdisk_partition *cur = NULL;
+    struct fdisk_iter *itr = NULL;
+    guint64 start = 0;
+    gboolean found = FALSE;
+
+    itr = fdisk_new_iter (FDISK_ITER_FORWARD);
+    if (!itr) {
+        g_set_error (error, BD_PART_ERROR, BD_PART_ERROR_FAIL,
+                     "Failed to create a new iterator");
+        return FALSE;
+    }
+
+    cur = fdisk_table_get_partition_by_partno (tb, partno);
+    if (!cur) {
+        g_set_error (error, BD_PART_ERROR, BD_PART_ERROR_FAIL,
+                     "Failed to locate partition '%d' in table.", partno);
+        fdisk_free_iter (itr);
+        return FALSE;
+    }
+
+    start = fdisk_partition_get_start (cur);
+
+    while (fdisk_table_next_partition (tb, itr, &pa) == 0) {
```

Comment:
I'd prefer `!found && fdisk_table_next_partition())` here and getting rid of the `break`s below.

Reply:
Ok, you're the boss here :-)